### PR TITLE
ENG-867: Introduce build script

### PR DESCRIPTION
**Pull request by Sudokamikaze**

### CODE REVIEW

**Comment by Sudokamikaze** [comment]
@EvgeniyPatlan @vorsel please review

File: patchelf.spec [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+make DESTDIR=$RPM_BUILD_ROOT install
+# rpmbuild automatically strips... strip $RPM_BUILD_ROOT/%%{_bindir}/* || true
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%files
+%{_bindir}/patchelf
+%doc %{_docdir}/patchelf/README
+%{_mandir}/man1/patchelf.1.gz
```
Comment (EvgeniyPatlan) [inline patchelf.spec:44]: please add changelog

File: patchelf_builder.sh [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+        --get_sources       Source will be downloaded from github
+        --build_src_rpm     If it is 1 src rpm will be built
+        --build_source_deb  If it is 1 source deb package will be built
+        --build_rpm         If it is 1 rpm will be built
+        --build_deb         If it is 1 deb will be built
+        --install_deps      Install build dependencies(root previlages are required)
+        --branch            Branch for build
+        --repo              Repo for build
+        --help) usage ;;
+Example $0 --builddir=/tmp/PXB --get_sources=1 --build_src_rpm=1 --build_rpm=1
```
Comment (EvgeniyPatlan) [inline patchelf_builder.sh]: PXB?

File: patchelf_builder.sh [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+        exit 1
+    fi
+    CURPLACE=$(pwd)
+    if [ "x$OS" = "xrpm" ]
+    then
+        yum -y install git wget 
+        if [[ "${RHEL}" -eq 8 ]]; then         
+            PKGLIST+=" binutils-devel libev-devel bison make gcc"
+            PKGLIST+=" rpm-build gcc-c++"
+            PKGLIST+=" vim-common rpmlint autoconf automake "
```
Comment (EvgeniyPatlan) [inline patchelf_builder.sh]: why do we need vim-common?

File: patchelf_builder.sh [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+            until yum -y install ${PKGLIST}; do
+                echo "waiting"
+                sleep 1
+            done
+        fi
+    else
+        apt-get update
+        DEBIAN_FRONTEND=noninteractive apt-get -y install lsb-release gnupg git wget
+
+        PKGLIST+=" bison make devscripts debconf debhelper automake bison autoconf automake"
```
Comment (EvgeniyPatlan) [inline patchelf_builder.sh]: don't you needed dpkg-dev to create deb package?

File: patchelf_builder.sh [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+    fi
+    ls | grep -v patchelf*.tar.* | xargs rm -rf
+    mkdir -vp rpmbuild/{SOURCES,SPECS,BUILD,SRPMS,RPMS}
+
+    TARFILE=$(basename $(find . -name 'patchelf-*.tar.gz' | sort | tail -n1))
+    NAME=$(echo ${TARFILE}| awk -F '-' '{print $1}')
+    VERSION_TMP=$(echo ${TARFILE}| awk -F '-' '{print $2}')
+    VERSION=${VERSION_TMP%.tar.gz}
+    #
+    git clone https://github.com/Sudokamikaze/patchelf-packaging.git
```
Comment (EvgeniyPatlan) [inline patchelf_builder.sh]: why it is not included into source tarball?

File: patchelf_builder.sh [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+    NAME=$(echo ${TARFILE}| awk -F '-' '{print $1}')
+    VERSION_TMP=$(echo ${TARFILE}| awk -F '-' '{print $2}')
+    VERSION=${VERSION_TMP%.tar.gz}
+
+    echo "DEB_RELEASE=${DEB_RELEASE}" >> ${CURDIR}/patchelf.properties
+
+    NEWTAR=${NAME}_${VERSION}.orig.tar.gz
+    mv ${TARFILE} ${NEWTAR}
+
+    git clone https://github.com/Sudokamikaze/patchelf-packaging.git
```
Comment (EvgeniyPatlan) [inline patchelf_builder.sh]: it should be in source tarball

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Comment by Sudokamikaze** [comment]
@EvgeniyPatlan @vorsel

Please, re-review, as I've added patch for `https://github.com/NixOS/patchelf/issues/138`
`https://github.com/NixOS/patchelf/pull/127`

Which affects us too

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**